src/spatio_temporal_voxel_grid.cpp

```diff
@@ -228,11 +235,16 @@ void SpatioTemporalVoxelGrid::TemporalClearAndGenerateCostmap(                \
         {
           std::cout << "Failed to clear point." << std::endl;
         }
+        else
+        {
+          updated_cells.insert(occupany_cell(pose_world[0], pose_world[1]));
+        }
         continue;
       }
     }
     // if here, we can add to costmap and PC2
     PopulateCostmapAndPointcloud(pt_index);
+    updated_cells.insert(occupany_cell(pose_world[0], pose_world[1]));
```

Comment:
This represents the case where a point currenctly in the grid stays in the grid, this doesn't need to have the updated cells applied to it -- there's no change in this situation

Reply:
The variable `updated_cells` really should be `cleared_cells` because its not including the newly marked cells, only the newly removed ones.

Reply:
I agree that this represents the case where a point currently in the grid stays in the grid. However, because this is the only place where the flattened costmap is updated with obstacle information, it ends up representing all obstacles (i.e. marked cells) of the current cycle - which is why I store it in the `updated_cells` variable.

Please see my answer below for more context on this.

Reply:
You don't have to though, nothing has changed to need to be touched. Plus since this populates the flattened costmap, the `UpdateROSCostmap` function will take care of touching the cell for you https://github.com/SteveMacenski/spatio_temporal_voxel_layer/blob/ros2/src/spatio_temporal_voxel_layer.cpp#L688

Reply:
Fixed